File: examples/java/kora-java-http-server/src/test/java/ru/tinkoff/kora/example/httpserver/TestApplication.java
package ru.tinkoff.kora.example.httpserver;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;
import ru.tinkoff.kora.example.http.server.Application;
import ru.tinkoff.kora.example.http.server.JsonPostController;
import ru.tinkoff.kora.http.client.common.annotation.HttpClient;
import ru.tinkoff.kora.http.client.jdk.JdkHttpClientModule;
import ru.tinkoff.kora.http.common.HttpMethod;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;
import ru.tinkoff.kora.json.common.annotation.Json;

/**
 * Тестовый контейнер приложение, который расширяет основное приложение и например добавляет
 * некоторые компоненты
 * из общих модулей, которые не используются в данном приложении, которые могут быть например
 * использованы в других подобных приложениях.
 * Например, когда у вас есть разные приложения READ API и WRITE API.
 * Либо, вам нужны некоторые функции сохранения/удаления/обновления только для тестирования в
 * качестве быстрой тестовой утилиты.
 * <p>
 * Но мы НАСТОЯТЕЛЬНО РЕКОМЕНДУЕМ ТЕСТИРОВАТЬ приложения как черный ящик в качестве основного
 * подхода.
 * -------
 * Test Application than extends Real Application and may be adds some components
 * from common modules that are not used in Real App, but may be used in other similar apps.
 * Like when you have different READ API application and WRITE API application
 * or may be, you need some save/delete/update methods only for testing as fast test utils.
 * <p>
 * But we STRONGLY ENCOURAGE AND RECOMMEND TO USE black box testing as a primary source of truth for
 * tests.
 */
@KoraApp
public interface TestApplication extends Application, JdkHttpClientModule {

    @Root
    @Component
    @HttpClient(configPath = "testHttpClient")
    interface JsonHttpClient {

        @HttpRoute(method = HttpMethod.POST, path = "/json")
        @Json
        JsonPostController.JsonResponse post(@Json JsonPostController.JsonRequest body);
    }
}

File: examples/kotlin/kora-kotlin-http-server/src/test/kotlin/ru/tinkoff/kora/kotlin/example/httpserver/TestApplication.kt
package ru.tinkoff.kora.kotlin.example.httpserver

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.http.client.common.annotation.HttpClient
import ru.tinkoff.kora.http.client.jdk.JdkHttpClientModule
import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.HttpRoute
import ru.tinkoff.kora.json.common.annotation.Json
import ru.tinkoff.kora.kotlin.example.http.server.Application
import ru.tinkoff.kora.kotlin.example.http.server.JsonPostController

@KoraApp
interface TestApplication : Application, JdkHttpClientModule {

    @Root
    @Component
    @HttpClient(configPath = "testHttpClient")
    interface JsonHttpClient {

        @HttpRoute(method = HttpMethod.POST, path = "/json")
        @Json
        fun post(@Json body: JsonPostController.JsonRequest): JsonPostController.JsonResponse
    }
}
